src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -236,10 +232,8 @@ class AutoImportFix(element: RsElement) : LocalQuickFixOnPsiElement(element), Hi
                     is TraitImplSource.Object -> return null
                 }
 
-                if (lookup.isTraitVisibleFrom(trait, methodCall)) return null
-                traitsToImport += trait
-            }
-            return traitsToImport
+                trait
+            }.filterInScope(methodCall)
```

Comment:
Here is the main wrong place
You need something like the following:
```kotlin
val traits = resolveResults.mapNotNull { variant -> ... }
return if (traits.filterInScope(methodCall).isNotEmpty()) null else traits
```
If there is at least one trait in scope, we should return `null` because it means that method can be resolved by rustc, otherwise return all traits as initial import candidates